subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DirectorySnapshotterTest.groovy

```diff
@@ -71,6 +72,16 @@ class DirectorySnapshotterTest extends Specification {
         def actuallyFiltered = new AtomicBoolean(false)
         when:
         def snapshot = directorySnapshotter.snapshot(rootDir.absolutePath, null, actuallyFiltered)
+
+        then:
+        ! actuallyFiltered.get()
+
+        1 * statisticsCollector.recordVisitHierarchy()
```

Comment:
I think it would be better to use a `Stub` for most of the tests and then test the statistics collection separately.

Reply:
Removed.

Reply:
OK, changed it around.